pyk/src/pyk/proof/proof.py

```diff
@@ -303,31 +314,169 @@ def lines(self) -> list[str]:
         return [line for lines in (summary.lines for summary in self.summaries) for line in lines]
 
 
-class ProofStep: ...
+class FailureInfo: ...
 
 
-class StepResult: ...
+class ProverPool(ContextManager['ProverPool'], Generic[P, PS, SR]):
```

Comment:
This class seems to belong to the implementation of `parallel_advance_proof`, consider sliding it below that function. Consider making it private even.

Reply:
Fixed.